packages/target-tools/src/telemetryProvider.js

```diff
@@ -28,13 +36,31 @@ export function TelemetryProvider(
     telemetryEntries.push({
       requestId,
       timestamp,
+      mode,
       features: {
-        decisioningMethod
+        decisioningMethod,
+        executePageLoad: isExecutePageLoad(request),
+        executeMboxCount: executeMBoxCount(request),
+        prefetchPageLoad: isPrefetchPageLoad(request),
+        prefetchMboxCount: prefetchMBoxCount(request),
+        prefetchViewCount: prefetchViewCount(request)
       },
       ...entry
     });
   }
 
+  function setMode(decisioningMethod) {
```

Comment:
Apologies, but I realized I misunderstood the flow for how Hybrid/ODD requests get resolved if the existing artifact doesn't have the decisioning rules to satisfy the request.  So if `on-device-decisioning` is set, and we are unable to resolve locally then there is no fallback and don't send any offers back.  If `hybrid` is set, then we would fallback to doing a Delivery API request and NOT download a new artifact.  

So that means we don't need to add a new subscription to the ArtifactProvider.  What we actually need to do is use the `response.status` to determine if the request was resolved locally or via Delivery API for hybrid requests.
`if (decisioningMethod === DECISIONING_METHOD.HYBRID && response.status === PARTIAL_CONTENT ) {
  mode = EXECUTION_MODE.ON_DEVICE;
} else if (decisioningMethod === DECISIONING_METHOD.HYBRID && response.status === OK ) {
  mode = EXECUTION_MODE. ON_DEVICE_CACHED;
}`

So we should rename setMode to `getMode` and add this ^ logic into it.  And also pass response.status into addEntry, where we'll call getMode().  Mode no longer needs to be a top-level member variable inside TelemetryProvider since we'll always determine the mode inside addEntry.

Sorry for the confusion on that and let me know if anything isn't clear.

Reply:
That sounds a LOT easier so I'm happy with that haha